app/services/create_or_update_location.rb

```diff
@@ -58,8 +55,12 @@ def build_new_version(params)
   end
 
   def create_new_version(params)
+    guiders = location.guiders
     build_new_version(params)
     @location.save!
+    guiders.each do |guider|
```

Comment:
I wonder if it just makes sense to just `update_all(location_id: X)` here instead of copying? Is there a case when prior versions are restored?